tools/run-mainnet.sh

```diff
@@ -1,22 +1,48 @@
-NAMESPACE=main
+#!/bin/bash
+# Copyright (c) 2022 The MobileCoin Foundation
+NAMESPACE=prod
+NET=main
 
-WORK_DIR="$HOME/.mobilecoin/${NAMESPACE}"
+WORK_DIR="$HOME/.mobilecoin/${NET}"
 WALLET_DB_DIR="${WORK_DIR}/wallet-db"
 LEDGER_DB_DIR="${WORK_DIR}/ledger-db"
-mkdir -p ${WORK_DIR}
+INGEST_DOWNLOAD_LOCATION="$WORK_DIR/ingest-enclave.css"
 
-(cd ${WORK_DIR} && CONSENSUS_SIGSTRUCT_URI=$(curl -s https://enclave-distribution.${NAMESPACE}.mobilecoin.com/production.json | grep consensus-enclave.css | awk '{print $2}' | tr -d \" | tr -d ,)
-curl -O https://enclave-distribution.${NAMESPACE}.mobilecoin.com/${CONSENSUS_SIGSTRUCT_URI})
+# Pass "--no-build" if the user just wants to run what they have in  
+# WORK_DIR instead of building and copying over a new exectuable
+if [ $# -eq 0 ] || [ $1 != "--no-build" ]; then
+    echo "Building"
+    mkdir -p ${WORK_DIR}
+    SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+    (cd ${WORK_DIR} && INGEST_SIGSTRUCT_URI=$(curl -s https://enclave-distribution.${NAMESPACE}.mobilecoin.com/production.json | grep ingest-enclave.css | awk '{print $2}' | tr -d \" | tr -d ,)
+    curl -O https://enclave-distribution.${NAMESPACE}.mobilecoin.com/${INGEST_SIGSTRUCT_URI})
+    INGEST_ENCLAVE_CSS=$INGEST_DOWNLOAD_LOCATION
+
+    $SCRIPT_DIR/build-fs.sh $NET
+    cp $SCRIPT_DIR/../target/release/mc-full-service $WORK_DIR
+fi
+
+
+if [ -z "$INGEST_ENCLAVE_CSS" ] ; then
+    if ! test -f "$INGEST_DOWNLOAD_LOCATION"; then
+        (cd ${WORK_DIR} && INGEST_SIGSTRUCT_URI=$(curl -s https://enclave-distribution.${NAMESPACE}.mobilecoin.com/production.json | grep ingest-enclave.css | awk '{print $2}' | tr -d \" | tr -d ,)
+        curl -O https://enclave-distribution.${NAMESPACE}.mobilecoin.com/${INGEST_SIGSTRUCT_URI})
+        INGEST_ENCLAVE_CSS=$INGEST_DOWNLOAD_LOCATION
+    else
+        echo "If running with --no-build, please export a path for the fog-ingest-enclave-css as INGEST_ENCLAVE_CSS"
```

Comment:
So it looks like if I run the script twice in a row, the behavior would be different. I haven't tested it locally but what I think is going to happen if INGEST_ENCLAVE_CSS is empty, the first time it will download the file because INGEST_DOWNLOAD_LOCATION doesn't exist, but the second time I will get this error since INGEST_DOWNLOAD_LOCATION was created. Is that right or am I missing something?

Reply:
Whats the reason to not have this run script right now?
Looks like a script that differentiates between a few networks would not be substantially more work.

Separately, on the topic of the ingest CSS file. I think it's fine downloading if it doesn't exist and we know where to get it from. I don't think it makes sense for the script to behave differently between runs, where one succeeds and the other one fails due to this else clause.